src/backend/distributed/executor/local_executor.c

```diff
@@ -99,42 +99,40 @@
 #include "nodes/params.h"
 #include "utils/snapmgr.h"
 
-
 /* controlled via a GUC */
 bool EnableLocalExecution = true;
 bool LogLocalCommands = false;
 
 bool TransactionAccessedLocalPlacement = false;
 bool TransactionConnectedToLocalGroup = false;
 
-
 static void SplitLocalAndRemotePlacements(List *taskPlacementList,
 										  List **localTaskPlacementList,
 										  List **remoteTaskPlacementList);
-static uint64 ExecuteLocalTaskPlan(CitusScanState *scanState, PlannedStmt *taskPlan,
-								   char *queryString);
+static uint64 ExecuteLocalTaskPlan(PlannedStmt *taskPlan, char *queryString,
+								   Tuplestorestate *tupleStoreState, ParamListInfo
+								   paramListInfo);
 static void LogLocalCommand(Task *task);
 static void ExtractParametersForLocalExecution(ParamListInfo paramListInfo,
 											   Oid **parameterTypes,
 											   const char ***parameterValues);
 static void LocallyExecuteUtilityTask(const char *utilityCommand);
 static void LocallyExecuteUdfTaskQuery(Query *localUdfCommandQuery);
 
-
 /*
- * ExecuteLocalTasks gets a CitusScanState node and list of local tasks.
+ * ExecuteLocalTasks executes the given tasks locally.
  *
  * The function goes over the task list and executes them locally.
- * The returning tuples (if any) is stored in the CitusScanState.
+ * The returning tuples (if any) is stored in the tupleStoreState.
  *
  * The function returns totalRowsProcessed.
  */
 uint64
-ExecuteLocalTaskList(CitusScanState *scanState, List *taskList)
+ExecuteLocalTaskList(List *taskList, ParamListInfo orig_paramListInfo,
```

Comment:
`ExecuteLocalTaskList `  does way too many things than I expected to see (like using cached plans or handling parameters etc). For example, see  `ExecuteTaskList`, which has a  pretty simple API and does something  pretty simple.

Are we planning  to use  `ExecuteLocalTaskList ` in any other PR? If so, it might make sense  to have `ExecuteLocalTaskListExtended`, which actually gets all the  necessary parameters, and the existing call should use `ExecuteLocalTaskListExtended`.

For other calls, we should have `ExecuteLocalTaskList `, where we simply pass those fields as `NULL`, but only pass `List *taskList` and maybe `Tuplestorestate *tupleStoreState`. (Hopefully, we can use that function inside of `ExecuteLocalUtilityTaskList ` as well)

Does that make sense?

Reply:
Can you check the PR again @onderkalaci, is this what you meant?